frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import {
+    Box,
+    CircularProgress,
+    CircularProgressProps,
+    LinearProgress,
+    LinearProgressProps,
+    Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface ProgressBarProps {
+    linear?: boolean; //by default - false
+    showProgress?: boolean; //by default - false
+    value?: number; //progress value
+    defaultValue?: number; //default progress value
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear, showProgress, value, defaultValue } = props;
```

Comment:
no need to destructure value and defaultValue

Reply:
But, without destructuring, the `value` and `defaultValue`, how we'll use it with the `props`? Because, in the `Button.tsx` file, similar thing was done, the desturucturing one (that's why I took some help from there).

Reply:
I never said the current code is all clean :-D

Reply:
you're using the props.value already and it should only be used there ...
`const val = useDynamicProperty(props.value, props.defaultValue, undefined);`

Reply:
So, do I've to remove both the `value` and the `defaultValue` params from the props?